src/zope/file/contenttype.rst

```diff
@@ -2,16 +2,34 @@
 Content type and encoding controls
 ==================================
 
+..
+  cgi.FieldStorage in Python 3.4--3.6, at least, has a __del__
+  method that wants to close the BytesIO file it has. This plays badly
+  with zope.publisher.browser.FileUpload which copies methods from the
+  BytesIO, but *does not* keep a reference to the FieldStorage. This
+  means that it's impossible to successfully upload file data without
+  getting "ValueError: I/O operation on closed file". We
+  remedy this by removing the offending method.
+
+  This was fixed in zope.publisher 4.3.1. See
+  https://github.com/zopefoundation/zope.publisher/pull/13
+
+  >>> import cgi
+  >>> try:
+  ...     del cgi.FieldStorage.__del__
+  ... except AttributeError:
+  ...     pass
```

Comment:
This will also be removed once zope.publisher 4.3.1 is out, yes?  And our setup.py will insist on having zope.publisher >= 4.3.1, yes?